### DISCUSSION TITLE
Binding and Uno.Microsoft.Xaml.Behaviors.WinUI.Managed in different projects

### DISCUSSION THREAD

**Discussion by leaderanalytics** [unoplatform/uno#11321] [Q&A] [answered]
I have two projects  in my solution.  In one project - a library project I added, `Binding` in XAML files builds and works well.  In the other project - provided by the new project template, `Binding` generates compile time errors and only `x:Bind` works.  I eventually want to switch everything to `x:Bind` but until that happens I am interested to know why `Binding` works in one project and not the other.  

Library project I added: `<Project Sdk="MSBuild.Sdk.Extras/3.0.23">` `Binding` works, `Uno.Microsoft.Xaml.Behaviors.WinUI.Managed` works

Default library project added by the template: `<Project Sdk="Microsoft.NET.Sdk">` `Binding` does not work, `Uno.Microsoft.Xaml.Behaviors.WinUI.Managed` does not work

> Error	XLS0505	Type 'Binding' is used like a markup extension but does not derive from MarkupExtension.	

I am referencing [Uno.Microsoft.Xaml.Behaviors.WinUI.Managed](https://github.com/unoplatform/Uno.XamlBehaviors) however my project does not recognize it.  [This article](https://nicksnettravels.builttoroam.com/winui-uno-behaviors/) does not indicate that a conditional package reference is required.  The [samples project](https://github.com/unoplatform/Uno.XamlBehaviors/blob/uno/samples/CS/XAMLBehaviorsSample/XAMLBehaviorsSample.csproj) references the source projects directly so I can't tell from that if I'm referencing it correctly.  Also, I am referencing same nuget in my library project and it works fine.

	<ItemGroup>
		//...
		<PackageReference Include="Uno.Microsoft.Xaml.Behaviors.WinUI.Managed" Version="2.3.1-uno.2" />  
	</ItemGroup>

XAML:

	xmlns:Interactivity="using:Microsoft.Xaml.Interactivity" 
    xmlns:Interactions="using:Microsoft.Xaml.Interactions.Core"
	
	//...
	
	
	<ComboBox Width="120" BorderBrush="Black" BorderThickness=".5" Padding="6" VerticalAlignment="Center">
		<Interactivity:Interacton.Behaviors>
			<Interactions:EventTriggerBehavior EventName="SelectionChanged">
				<Interactions:InvokeCommandAction Command="{x:Bind SelectedThemeChangedCommand}" />
			</Interactions:EventTriggerBehavior>
		</Interactivity:Interacton.Behaviors>
	</ComboBox>
	
	
	
	
> Error	WMC0010	Unknown attachable member 'Interacton.Behaviors' on element 'ComboBox'	
> Error	UXAML0001	The type {using:Microsoft.Xaml.Interactivity}Interacton could not be found	LeaderPivot.XAML.UNO.Host	
	
How can I use `Uno.Microsoft.Xaml.Behaviors.WinUI.Managed` in this project?

**Comment by jeromelaban** [comment]
This article is outdated and conditional references are needed.

You'll need something like this: 
```xml
<ItemGroup Condition="'$(TargetFramework)' == 'net6.0-windows10.0.18362.0'">
  <PackageReference Include="Microsoft.Xaml.Behaviors.WinUI.Managed" Version="2.0.9" />
</ItemGroup>
<ItemGroup Condition="'$(TargetFramework)' != 'net6.0-windows10.0.18362.0'">
  <PackageReference Include="Uno.Microsoft.Xaml.Behaviors.WinUI.Managed" Version="2.3.0" />
</ItemGroup>
```

**Comment by leaderanalytics** [comment, marked as answer]
Thank you @jeromelaban.  I added the references as specified and I still get the same errors.  I deleted bin/obj/.vs several times.

	<ItemGroup Condition="'$(TargetFramework)'=='net7.0-windows10.0.18362'">
		<PackageReference Include="Microsoft.Xaml.Behaviors.WinUI.Managed" Version="2.0.9" /> 
	
	
	
	<ItemGroup Condition="'$(TargetFramework)'!='net7.0-windows10.0.18362'">	
		<PackageReference Include="Uno.Microsoft.Xaml.Behaviors.WinUI.Managed" Version="2.3.1-uno.2" />
		
		
		
It appears that Uno.Community.Toolkit is referencing Uno.Microsoft.Xaml.Behaviors.WinUI.Managed 2.3.1-uno.2.  If I try to add 2.3.0 I get a downgrade error from nuget.

 
![image](https://user-images.githubusercontent.com/24495447/219078886-b6c6c31c-e380-4d5d-a197-a33528359d6c.png)

**Reply by jeromelaban** [reply]
If you have a repro project, can you provide it here?

**Reply by leaderanalytics** [reply]
Repo is [here](https://github.com/leaderanalytics/LeaderPivot.XAML.Uno).

**Reply by leaderanalytics** [reply]
@jeromelaban  I found my typo!  Very sorry!

**Reply by modula2019** [reply]
Hi @jeromelaban and @leaderanalytics, I know it is a long time ago this issue was raised. I am also trying to use interactivity with my combobox so that raise SelectionChanged in my viewmodel. I am using net8.0. When I added the itemgroup to target net8.0 windows, my application stopped debugging with a lot of errors. Is there no packages for net8.0 yet?

Please advise,
Best regards

 ```
<ItemGroup Condition="'$(TargetFramework)'=='net8.0-windows10.0.19041'">
   <PackageReference Include="Microsoft.Xaml.Behaviors.WinUI.Managed"/>
 </ItemGroup>

 <ItemGroup Condition="'$(TargetFramework)'!='net8.0-windows10.0.19041'">
   <PackageReference Include="Uno.Microsoft.Xaml.Behaviors.WinUI.Managed"/>
 </ItemGroup>
```

**Reply by jeromelaban** [reply]
@modula2019 it would be best to open another discussion with [the errors you're getting](https://platform.uno/docs/articles/uno-builds-troubleshooting.html).

**Reply by modula2019** [reply]
@jeromelaban I have opened another discussion here. Thanks
https://github.com/unoplatform/uno/discussions/16856

**Reply by agneszitte** [reply]
@modula2019, @jeromelaban I have left the details on https://github.com/unoplatform/uno/discussions/16856
